experimental/examples/bubblebench/tree/src/cursor/bubble.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IBubble } from "@fluid-example/bubblebench-common";
+import { Anchor, brand, FieldKey, ISharedTree } from "@fluid-internal/tree";
+import { SharedTreeNodeHelper } from "./tree-utils";
+
+export class Bubble implements IBubble {
+	static xFieldKey: FieldKey = brand("x");
+	static yFieldKey: FieldKey = brand("y");
+	static vxFieldKey: FieldKey = brand("vx");
+	static vyFieldKey: FieldKey = brand("vy");
+	static rFieldKey: FieldKey = brand("r");
+
+	private readonly treeNodeHelper: SharedTreeNodeHelper;
+
+	constructor(public readonly tree: ISharedTree, public readonly anchor: Anchor) {
+		this.treeNodeHelper = new SharedTreeNodeHelper(tree, anchor);
+	}
+
+	public get x() {
+		return this.treeNodeHelper.getFieldValue(Bubble.xFieldKey) as number;
+	}
+	public set x(value: number) {
+		this.treeNodeHelper.setFieldValue(Bubble.xFieldKey, value);
+	}
+
+	public get y() {
+		return this.treeNodeHelper.getFieldValue(Bubble.yFieldKey) as number;
```

Comment:
allocating a new cursor for every field access is really inefficient. This actually is slower than how it works in editable tree, which kind of defeats the purpose of this test/example.

TO make the comparison useful, Bubble should probably cache a cursor (and clear it before edits, and lazily restore in on read, like in EditableTree), and use that cursor (not a new one) to access field values.